packit/api.py

```diff
@@ -272,6 +272,7 @@ def sync_release(
             if not use_local_content:
                 self.up.local_project.git_repo.git.checkout(current_up_branch)
             self.dg.refresh_specfile()
+            self.dg.local_project.git_repo.git.reset("--hard")
```

Comment:
I'm confused. I expected this would be
```suggestion
            self.dg.local_project.git_repo.git.reset("--hard", f"remotes/origin/{dist_git_branch}")
```

Reply:
You're right, this is more correct, changed and added a test case